Instrumentation improvements

cole opened the pull request:
Connected to: https://github.com/archivematica/Issues/issues/964, 
Connected to: https://github.com/archivematica/Issues/issues/906

* update metric descriptions
* add new metrics for file format and modified timestamps
* zero mcpserver counters on startup
* switch timing metrics to histograms with defined buckets

This needs a few dashboard changes before it can be reviewed.

cole committed:
Update metric help text

cole committed:
Add new metrics for format count and modified time

cole committed:
Zero mcpserver counters on startup

cole committed:
Switch to histograms for timing

Histograms with defined buckets seem to be a better approach for
timing durations than summaries as the limitations are a bit clearer.

Also moves the archivematicaCommon metrics into their own module.

cole committed:
Improved histogram bucket breakdown

Fewer buckets, with a more standardized distribution and comments.

cole committed:
Use counters to track total time spent

These metrics are informational for developers. To be useful, we need
more accuracy than a summary provides.

cole committed:
Add counters for histogram values

Histograms give us the distribution of time spent, but for determining
bottlenecks and utilization it's useful to have the actual totals.

cole committed:
Revert "Add counters for histogram values"

This reverts commit 6ffc52aee980f3fc5b85036a909b6d446a6ea5f2.

cole committed:
Cleanup queries for file formats

These might be slower for small transfers now but should scale
to large transfers. Also group file group use into the ones we care
about to reduce cardinality.

cole committed:
Further reduce the number of histogram buckets

More is hard to visualize in Grafana, so not very useful.

cole committed:
Remove old chain_completed call

cole committed:
Use years, not timestamps for mod time buckets

cole committed:
Add transfer completed metrics

Number completed, file and bytes count, by transfer type

cole committed:
Switch to histograms so we can track distributions

cole committed:
Use the right histogram for transfer size

replaceafill commented on src/archivematicaCommon/lib/common_metrics.py:
```diff
@@ -0,0 +1,96 @@
+
+db_retry_time_counter = Counter(
+    "common_db_retry_time_seconds",
+    "Total time waiting to retry database transactions in seconds",
```
I've noticed you've been adding "labeled by" to some labels. Is it worth adding that here (and in the counter below) too?

replaceafill reviewed (approved):
LGTM! :+1:

cole committed:
Update common metrics description

cole committed:
30min = 1800 secs

cole committed:
Add 'Unknown' transfer type

Zipped and unzipped bags are two transfer types that never set a
value for transfer type in the database, resulting in a type of empty
string.

Since we can't differentiate between the two, just leave as 'Unknown'.
